code/modules/overmap/contacts/_contacts.dm

```diff
@@ -0,0 +1,103 @@
+#define SENSOR_TIME_DELAY 0.2 SECONDS
+/datum/overmap_contact
+
+	var/name  = "Unknown"            // Contact name.
+	var/image/marker                 // Image overlay attached to the contact.
+	var/pinged = FALSE				 // Used to animate overmap effects.
+	var/list/images = list()		 // Our list of images to cast to users.
+
+	// The sensor console holding this data.
+	var/obj/machinery/computer/ship/sensors/owner
+
+	// The actual overmap effect associated with this.
+	var/obj/effect/overmap/effect
+
+/datum/overmap_contact/New(var/obj/machinery/computer/ship/sensors/creator, var/obj/effect/overmap/source)
+	// Update local tracking information.
+	owner =  creator
+	effect = source
+	name =   effect.name
+	owner.contact_datums[effect] = src
+
+	marker = image(loc = effect, icon = effect.icon, icon_state = effect.icon_state)
+	marker.appearance_flags |= RESET_COLOR
+	marker.color = effect.color
+	marker.filters = filter(type="drop_shadow", color = marker.color + "F0", size = 2, offset = 1,x = 0, y = 0)
+	marker.alpha = 0 // Marker fades in on detection.	
+
+	images += marker
+
+/datum/overmap_contact/proc/update_marker_icon(var/range = 0)
+	marker.icon_state = effect.icon_state
+
+	marker.overlays.Cut()
+
+	if(check_effect_shield())
+		var/image/shield_image = image(icon = 'icons/obj/overmap.dmi', icon_state = "shield")
+		shield_image.pixel_x = 8
+		marker.overlays += shield_image
+
+	if(range > 1)
+		var/image/radar
+
+		for(var/image/I in images)
+			if(I.tag == "radar")
+				radar = I
+		if(radar)
+			return
+
+		radar = image(loc = effect, icon = 'icons/obj/overmap.dmi', icon_state = "sensor_range")
+		radar.pixel_x = -2
+		radar.tag = "radar"
+		radar.filters = filter(type="blur", size = 1)
+		images += radar
+
+		var/matrix/M = matrix()
+		M.Scale(range*2.6)
+		animate(radar, transform = M, alpha = 0, time = (SENSOR_TIME_DELAY*range), 1, SINE_EASING)
+		addtimer(CALLBACK(src, .proc/reset_radar, radar), (SENSOR_TIME_DELAY *range+0.1))
+		QDEL_IN(radar, (0.25 SECONDS *range+0.3))
```

Comment:
I'm unsure if qdeling images is even a thing, and even if it is, since tag is set, it'll never GC so that's a hard del every time I tihnk?
IMO should just reuse same image stored in a var instead of regenerating it every time.
just reset the alpha and scale and should be as good as new